### ISSUE TITLE
There is a problem with the dmp file

### ISSUE THREAD

**Issue by 24-2021** [tastypepperoni/PPLBlade#2]
Failed to obtain hash for dmp file

**Comment by tastypepperoni** [comment]
Are you running it on Windows 11?

**Comment by 24-2021** [comment]
> 您是否在 Windows 11 上运行它？

No, I'm in my virtual machine, Windows 10

**Comment by tastypepperoni** [comment]
Can you show the full command you ran?

**Comment by mamba-2021** [comment]
> 你能显示你运行的完整命令吗？

mimikatz # sekurlsa::minidump d.dmp
Switch to MINIDUMP : 'd.dmp'

mimikatz # sekurlsa::logonPasswords full
Opening : 'd.dmp' file for minidump...
ERROR kuhl_m_sekurlsa_acquireLSA ; Handle on memory (0x00000002)

**Comment by 24-2021** [comment]
> Can you show the full command you ran?

mimikatz # sekurlsa::minidump PPLBlade.dmp
Switch to MINIDUMP : 'PPLBlade.dmp'

mimikatz # sekurlsa::logonPasswords full
Opening : 'PPLBlade.dmp' file for minidump...
ERROR kuhl_m_sekurlsa_acquireLSA ; Modules informations

**Comment by mamba-2021** [comment]
> PPLBlade.dmp
There is no problem exporting PPLBlade.dmp, but I think decrypting hashes locally through mimikatz failed

**Comment by 24-2021** [comment]
> Can you show the full command you ran?

There is no problem exporting PPLBlade.dmp, but I think decrypting hashes locally through mimikatz failed

**Comment by tastypepperoni** [comment]
can you try to export hashes using Pypykatz and show the result?
pypykatz lsa minidump lsass.dmp

**Comment by tastypepperoni** [comment]
And also the full command of PPLBlade

**Comment by 24-2021** [comment]
> 您可以尝试使用 Pypykatz 导出哈希并显示结果吗？ Pypykatz LSA minidump LSASS.dmp

![image](https://github.com/tastypepperoni/PPLBlade/assets/75511051/dd646835-ddf8-4636-82b3-5cfd44b58cad)
![image](https://github.com/tastypepperoni/PPLBlade/assets/75511051/2b2b8b83-ae68-4da6-9a29-711ecb398d42)

I directly copied the. dmp generated by "PPLBlade. exe -- mode dothatlsassthing" and decrypted it to Kali using Pypykatz, but it still failed

**Comment by tastypepperoni** [comment]
Looks like it's the problem with pypykatz/mimikatz.
If you look at the section where it says "BASIC INFO", it seems that Pypykatz was able to retrieve some data from the dump file (OS information) but failed to parse the part where the hashes are stored.
Unfortunately, I'm unable to reproduce this issue in my VMs on freshly cloned and built versions of PPLBlade.
I faced a similar situation a couple of days ago. I don't actually know what fixed the problem, but after some time, Pypykatz started working again on my dump files. Probably something changed inside the lsass.exe process, that pypykatz wasn't able to handle before.
I suggest just restarting the system and testing it against the freshly started lsass.exe process.

**Comment by mamba-2021** [comment]
> 

ok, but I still failed after trying. I will change the system for retesting later when I have time

**Comment by 24-2021** [comment]
> 看起来这是pypykatz/mimikatz的问题。如果您查看显示“基本信息”的部分，似乎 Pypykatz 能够从转储文件（操作系统信息）中检索一些数据，但未能解析存储哈希的部分。不幸的是，我无法在新克隆和构建的 PPLBlade 版本的 VM 中重现此问题。几天前我遇到了类似的情况。我实际上不知道是什么解决了这个问题，但一段时间后，Pypykatz 再次开始处理我的转储文件。可能是 lsass.exe 流程内部发生了一些变化，这是 pypykatz 以前无法处理的。我建议重新启动系统并针对新启动的 lsass.exe 进程对其进行测试。

ok, but I still failed after trying. I will change the system for retesting later when I have time

**Comment by tastypepperoni** [comment]
Please let me know about the result of retest

**Comment by 24-2021** [comment]
> Please let me know about the result of retest

ok, I'll leave a message for you

**Comment by DrorDvash** [comment]
same for me. same error with mimikatz and pypykatz.

also, i cannot dump lsass again on the same machine using this tool.

![image](https://github.com/tastypepperoni/PPLBlade/assets/8413651/8f07375f-976a-4405-bcdc-2a5519682d94)


tried few times with different modes, and cleanup.

**Comment by tastypepperoni** [comment]
It seems that procexp driver is already running on the system. Here are your options:
1) run it in cleanup mode, if you manually terminated the execution previously. Note that options of the cleanup should be the same, as for the execution you're trying to clean up. You can also manually stop snd delete the service: sc stop PPLBlade; sc delete PPLBlade(assuming you did not change the default service name option)
2) if you have process explorer running on the system, it will come into conflict with PPLBlade's service. Close process explorer and stop the service(or just restart the system)

**Comment by DrorDvash** [comment]
i already ran cleanup with the same arguments. stopping and deleting the service did not helped also.
![image](https://github.com/tastypepperoni/PPLBlade/assets/8413651/3bdec77b-f955-48bc-b6cc-bade23380abd)

yes, i can try to restart the machine, but this is a smaller problem. I cannot get the hashes with mimikatz as this issue is all about :)

**Comment by tastypepperoni** [comment]
Unfortunately, I can not reproduce the issue of getting hashes from a dump file.
The tool has been tested on various kinds of hosts, with different OSes and different defense configurations, before the release.
The dump file, generated by the tool, has also been tested with different versions of mimikatz/pypykatz.
I just tested the tool again on a couple of hosts, and all seems to be working fine.
This is my last shot, what is the architecture type of the host, you're running the tool on? (32bit, 64bit). The tool has only been tested on 64-bit architectures.
If that's not the problem as well, I'm afraid that's all I was able to offer as help and we will have to close the issue as "close as not planned", since the problem does not seem to be about the tool.
